recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,352 @@
+ CMakeLists.txt                               |  30 +----
+ src/CMakeLists.txt                           | 190 ++++++++++++---------------
+ src/third_party/curl/CMakeLists.txt          |  16 +--
+ src/third_party/hdr_histogram/CMakeLists.txt |  15 +--
+ 4 files changed, 97 insertions(+), 154 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c6150948b..6d9f13965 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,6 +1,3 @@
+-cmake_minimum_required(VERSION 2.8.12)
+-project(cassandra C CXX)
+-
+ set(CASS_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+ set(CASS_SRC_DIR "${CASS_ROOT_DIR}/src")
+ set(CASS_INCLUDE_DIR "${CASS_ROOT_DIR}/include")
+@@ -29,8 +26,6 @@ endif()
+ 
+ option(CASS_BUILD_EXAMPLES "Build examples" OFF)
+ option(CASS_BUILD_INTEGRATION_TESTS "Build integration tests" OFF)
+-option(CASS_BUILD_SHARED "Build shared library" ON)
+-option(CASS_BUILD_STATIC "Build static library" OFF)
```

Comment:
Isn't it easier to do the following in the conanfile?
```python
self._cmake.definitions["CASS_BUILD_SHARED"] = self.options.shared
self._cmake.definitions["CASS_BUILD_STATIC"] = not self.options.shared
```

Reply:
It is BUT we do not want to use custom variables for shared/static builds, and we do not want to define two targets in cmake, because it is not a good practice since we have BUILD_SHARED_LIBS cmake variable to control this behavior.

Reply:
You're right about best practices.
But by rewriting, you effectively fork the cmake build script.
This is unnecessary extra work for no gain because conan already acts as an abstraction layer between cassandra and consumers.
Conan consumers won't face it anyways.

Reply:
[Fixed](https://github.com/conan-io/conan-center-index/pull/3110/commits/eaa640430ae6008d0df8a0b86d894f17ca4e98da)